providers/src/airflow/providers/google/cloud/operators/pubsub.py

```diff
@@ -774,32 +782,61 @@ def __init__(
         self.ack_messages = ack_messages
         self.messages_callback = messages_callback
         self.impersonation_chain = impersonation_chain
+        self.deferrable = deferrable
+        self.poll_interval = poll_interval
 
     def execute(self, context: Context) -> list:
-        hook = PubSubHook(
-            gcp_conn_id=self.gcp_conn_id,
-            impersonation_chain=self.impersonation_chain,
-        )
-
-        pulled_messages = hook.pull(
-            project_id=self.project_id,
-            subscription=self.subscription,
-            max_messages=self.max_messages,
-            return_immediately=True,
-        )
-
-        handle_messages = self.messages_callback or self._default_message_callback
-
-        ret = handle_messages(pulled_messages, context)
+        if self.deferrable:
+            self.defer(
+                trigger=PubsubPullTrigger(
+                    subscription=self.subscription,
+                    project_id=self.project_id,
+                    max_messages=self.max_messages,
+                    ack_messages=self.ack_messages,
+                    gcp_conn_id=self.gcp_conn_id,
+                    poke_interval=self.poll_interval,
+                    impersonation_chain=self.impersonation_chain,
+                ),
+                method_name=GOOGLE_DEFAULT_DEFERRABLE_METHOD_NAME,
+            )
+        else:
```

Comment:
It's not necessary to use `else`. When the code goes to the `defer` mode after that it never returns to `execute` method.